src/daemon/daemon.js

```diff
@@ -43,18 +41,28 @@ async function cleanup (ipfsd) {
   }
 }
 
-async function spawn ({ type, path, keysize }) {
-  const factory = IPFSFactory.create({ type: type })
+async function spawn ({ type, path, flags, keysize }) {
+  // NOTE: presence of env variable IPFS_PATH overrides config
+  path = process.env.IPFS_PATH || path
 
-  const ipfsd = await factory.spawn({
+  const factory = Ctl.createFactory({
+    remote: false,
     disposable: false,
-    defaultAddrs: true,
-    repoPath: path,
+    test: false,
+    args: flags,
+    type: type
+  })
+
+  const ipfsd = await factory.spawn({
+    ipfsOptions: {
+      repo: path
+    },
```

Comment:
I believe basic init did not work because repo path got moved to `ipfsOptions`.  This was the fix :)

Reply:
About the `api` file: I thought IPFS itself handled it! The only thing we did was this: #722.